app/api.py

```diff
@@ -654,26 +665,44 @@ def post(self):
                 style='plain')
             return
 
+        query_lang = None
+        query_action = None
+        match = None
+        for lang, action, regex in HandleSMS.QUERY_PATTERNS:
+            match = re.search(regex, message_text.strip(), re.I)
+            if match:
+                query_lang = lang
+                query_action = action
+                break
+
+        if query_lang:
+            # Use the language for the following calls of _().
+            django_setup.activate(query_lang)
+
         responses = []
-        search_m = re.search(r'^search\s+(.+)$', message_text.strip(), re.I)
-        add_self_m = re.search(r'^i am\s+(.+)$', message_text.strip(), re.I)
-        if search_m:
-            query_string = search_m.group(1).strip()
+
+        if query_action == 'search':
+            query_string = match.group(1).strip()
             query = TextQuery(query_string)
             persons = indexing.search(repo, query, HandleSMS.MAX_RESULTS)
             if persons:
                 for person in persons:
                     responses.append(self.render_person(person))
             else:
-                responses.append('No results found for: %s' % query_string)
+                responses.append(
+                    _('No results found for: %(query)s')
+                        % {'query': query_string})
             responses.append(
-                'More at: google.org/personfinder/%s?ui=light' % repo)
+                _('More at: %(url)s')
+                    % {'url': 'google.org/personfinder/%s?ui=light' % repo})
             responses.append(
-                'All data entered in Person Finder is available to the public '
-                'and usable by anyone. Google does not review or verify the '
-                'accuracy of this data google.org/personfinder/global/tos')
-        elif self.config.enable_sms_record_input and add_self_m:
-            name_string = add_self_m.group(1).strip()
+                _('All data entered in Person Finder is available to the '
+                  'public and usable by anyone. Google does not review or '
+                  'verify the accuracy of this data '
```

Comment:
I don't know why. Maybe we wanted to keep it as shorter as possible :) I prefer to keep it as is in this change because it was like this before this change.